src/components/App/App.jsx

```diff
@@ -165,99 +225,105 @@ function App() {
-      <CurrentPageContext.Provider value={currentPage}>
-        <div className="page">
-          <div className="page__content">
-            <Routes>
-              <Route
-                path="/"
-                element={
-                  <>
-                    <Header
-                      handleLoginClick={handleLoginClick}
-                      onLogout={onLogout}
-                      handleMenuClick={handleMenuClick}
-                      isOpen={activeModal !== ""}
-                      onSearch={onSearch}
-                    />
-                    <Main
-                      handleLoginClick={handleLoginClick}
-                      handleSaveItem={handleSaveItem}
-                      handleRemoveSave={handleRemoveSave}
-                      searchResults={searchResults}
-                      keyword={keyword}
-                      isLoading={isLoading}
-                      searchError={searchError}
-                      savedItems={savedItems}
-                    />
-                  </>
-                }
-              />
+    <CurrentUserContext.Provider value={currentUser}>
+      <IsLoggedInContext.Provider value={isLoggedIn}>
+        <CurrentPageContext.Provider value={currentPage}>
+          <div className="page">
+            <div className="page__content">
+              <Routes>
+                <Route
+                  path="/"
+                  element={
+                    <>
+                      <Header
+                        handleLoginClick={handleLoginClick}
+                        onLogout={onLogout}
+                        handleMenuClick={handleMenuClick}
+                        isOpen={activeModal !== ""}
+                        onSearch={onSearch}
+                      />
+                      <Main
+                        handleLoginClick={handleLoginClick}
+                        handleSaveItem={handleSaveItem}
+                        handleRemoveSave={handleRemoveSave}
+                        searchResults={searchResults}
+                        keyword={keyword}
+                        isLoading={isLoading}
+                        searchError={searchError}
+                        savedItems={savedItems}
+                      />
+                    </>
+                  }
+                />
 
-              <Route
-                path="/saved-news"
-                element={
-                  <ProtectedRoute>
-                    <SavedNews
-                      onLogout={onLogout}
-                      savedItems={savedItems}
-                      handleRemoveSave={handleRemoveSave}
-                      handleMenuClick={handleMenuClick}
-                      isOpen={activeModal !== ""}
-                    />
-                  </ProtectedRoute>
-                }
+                <Route
+                  path="/saved-news"
+                  element={
+                    <ProtectedRoute isLoggedInLoading={isLoggedInLoading}>
```

Comment:
* The `Saved articles` page must be **protected** from unauthorized visits https://snipboard.io/m2iMK1.jpg . I can visit the `/saved-news` route

Check, please, why I'm able to visit the protected page without authorization